cmd/vic-machine/create/create.go

```diff
@@ -610,9 +610,14 @@ func (c *Create) processCertificates() error {
 	}
 
 	// if we've not got a specific CommonName but do have a static IP then go with that.
-	if c.cname == "" && c.clientNetworkIP != "" {
-		c.cname = c.clientNetworkIP
-		log.Infof("Using client-network-ip as cname where needed - use --tls-cname to override: %s", c.cname)
+	if c.cname == "" {
+		if c.clientNetworkIP != "" {
+			c.cname = c.clientNetworkIP
+			log.Infof("Using client-network-ip as cname where needed - use --tls-cname to override: %s", c.cname)
+		} else if c.publicNetworkIP != "" && (c.publicNetworkName == c.clientNetworkName || c.clientNetworkName == "") {
```

Comment:
What if the management ip is set only? The configuration are shared between networks if the port group is shared, so that should be reflected here as well.